lib/matplotlib/tests/test_sphinxext.py

```diff
@@ -97,10 +97,33 @@ def plot_directive_file(num):
     assert filecmp.cmp(range_6, plot_file(5))
 
 
-def build_sphinx_html(source_dir, doctree_dir, html_dir):
+def test_plot_html_show_source_link(tmpdir):
+    source_dir = Path(tmpdir) / 'src'
+    source_dir.mkdir()
+    shutil.copyfile(Path(__file__).parent / 'tinypages/conf.py', source_dir)
```

Comment:
https://github.com/matplotlib/matplotlib/pull/20814/checks?check_run_id=3318293981#step:12:185

```
E           IsADirectoryError: [Errno 21] Is a directory: '/tmp/pytest-of-runner/pytest-0/popen-gw1/test_plot_html_show_source_lin0/src'
```

It seems that the destination must be a file.

Reply:
I'll replace `shutil.copyfile` with `shutil.copy`.

Reply:
Oops, sorry about that.

Reply:
No problem!